lib/pure/net.nim

```diff
@@ -1702,14 +1714,42 @@ proc trySend*(socket: Socket, data: string): bool {.tags: [WriteIOEffect].} =
   ## and instead returns `false` on failure.
   result = send(socket, cstring(data), data.len) == data.len
 
+proc sendTo*(socket: Socket, address: IpAddress, port: Port, data: pointer,
+             size: int, af: Domain = AF_INET, flags = 0'i32): int {.
+              discardable, tags: [WriteIOEffect].} =
+  ## This proc sends `data` to the specified `IPAddress`. Hostnames
+  ## are not supported for this variant.
+  ##
+  ## If an error occurs an OSError exception will be raised. Otherwise
+  ## the number of bytes writen will be returned.
+  ##
+  ## **Note:** You may wish to use the high-level version of this function
+  ## which is defined below.
+  ##
+  ## **Note:** This proc is not available for SSL sockets.
+  assert(socket.protocol != IPPROTO_TCP, "Cannot `sendTo` on a TCP socket")
+  assert(not socket.isClosed, "Cannot `sendTo` on a closed socket")
+
+  var sa: Sockaddr_storage
+  var sl: Socklen
+  toSockAddr(address, port, sa, sl)
+
+  result = sendto(socket.fd, data, size.cint, flags.cint,
+                  cast[ptr SockAddr](addr sa), sl)
+
+  if result == -1'i32:
+    let osError = osLastError()
+    raiseOSError(osError)
+
 proc sendTo*(socket: Socket, address: string, port: Port, data: pointer,
-             size: int, af: Domain = AF_INET, flags = 0'i32) {.
-             tags: [WriteIOEffect].} =
+             size: int, af: Domain = AF_INET, flags = 0'i32): int {.
+             discardable, tags: [WriteIOEffect].} =
```

Comment:
A .discardable int still is a breaking change.

Reply:
Ah ok, yes it wasn't happy with the tests either. I'll revert that change but would it make sense to test the other one?